sysconfig/sysconfig.go

```diff
@@ -35,6 +39,28 @@ type Options struct {
 	// data, i.e. for cloud-init during the initramfs it will be something like
 	// boot.InstallHostWritableDir
 	TargetRootDir string
+
+	// GadgetDir is the path of the mounted gadget snap.
+	GadgetDir string
+}
+
+type FilesystemOnlyApplyOptions struct {
+	// Classic is true when the system in rootdir is a classic system
+	Classic bool
+}
+
+// ApplyFilesystemOnlyDefaultsImpl is initialized by init() of configcore.
+var ApplyFilesystemOnlyDefaultsImpl = func(rootDir string, defaults map[string]interface{}, options *FilesystemOnlyApplyOptions) error {
+	panic("ApplyFilesystemOnlyDefaultsImpl not set")
```

Comment:
is perhaps the reason the reason that this is unset here is to avoid a cyclic import path? if so perhaps we should expand the panic method to mention that configcore should have been imported too?